mindsdb_native/libs/controllers/predictor.py

```diff
@@ -53,6 +53,35 @@ def _prepare_sample_settings(user_provided_settings,
 
     return sample_settings, sample_function
 
+def _prepare_timeseries_settings(user_provided_settings):
+    timeseries_settings = dict(
+        is_timeseries=False
+        ,group_by=None
+        ,order_by=None
+        ,window=None
+        ,dynamic_window=None
+        ,use_previous_target=True
+        ,keep_order_column=True
+        ,nr_predictions=1
+    )
+
+    if len(user_provided_settings) > 0:
+        if 'order_by' not in user_provided_settings or ('window' not in user_provided_settings and 'dynamic_window' not in user_provided_settings):
```

Comment:
I'd split this `if` into two so it's easier to read the code and the exceptions are more specific.
```
if 'order_by' not in user_provided_settings:
    raise Exception('Invalid timeseries settings, please provide `order_by` key [a list of columns]')

if 'window' not in user_provided_settings and 'dynamic_window' not in user_provided_settings:
    raise Exception(f'Invalid timeseries settings, you must specify a window size with either `window` or `dynamic_window` key')

timeseries_settings['is_timeseries'] = True
```

Reply:
Made a refactor very similar to this, changed the `if`-ing with if/elif/else, but other than that it's the same.